crates/fuzzing/src/generators/single_inst_module.rs

```diff
@@ -0,0 +1,349 @@
+//! Generate Wasm modules that contain a single instruction.
+
+use arbitrary::{Arbitrary, Unstructured};
+use once_cell::sync::Lazy;
+use wasm_encoder::{
+    CodeSection, ExportKind, ExportSection, Function, FunctionSection, Instruction, Module,
+    TypeSection, ValType,
+};
+
+/// The name of the function generated by this module.
+const FUNCTION_NAME: &'static str = "test";
+
+/// Configure a single instruction module.
+///
+/// By explicitly defining the parameter and result types (versus generating the
+/// module directly), we can more easily generate values of the right type.
+#[derive(Clone, Debug)]
+pub struct SingleInstModule<'a> {
+    instruction: Instruction<'a>,
+    parameters: Vec<ValType>,
+    results: Vec<ValType>,
```

Comment:
I think it might be an improvement to avoid the use of `once_cell` in this module by making `parameters` and `results` be `&'a [ValType]` instead. (Or drop the `'a` lifetime parameter and just use `'static`.) The macros below would then use `&[...]` instead of `vec![...]`, and the instructions would be declared like:

```rust
static INSTRUCTIONS: &[SingleInstModule] = &[
    unary!(I32Clz, i32),
    ...
];
```

Reply:
Good suggestion! Done.